backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/invoice/C_Invoice_StepDef.java

```diff
@@ -144,6 +168,82 @@ public void generateInvoice(final int timeoutSec, @NonNull final DataTable table
+			final OrderId targetOrderId = OrderId.ofRepoId(orderRecord.getC_Order_ID());
+
+			orders.add(orderRecord);
+
+			//make sure the given invoice candidate is ready for processing
+			final Supplier<Boolean> noInvoiceCandidateRecompute = () ->
+			{
+				final IInvoiceCandDAO.InvoiceableInvoiceCandIdResult invoiceableInvoiceCandId = invoiceCandDAO.getFirstInvoiceableInvoiceCandId(targetOrderId);
+				return invoiceableInvoiceCandId.getFirstInvoiceableInvoiceCandId() != null;
+			};
+
+			StepDefUtil.tryAndWait(timeoutSec, 500, noInvoiceCandidateRecompute);
+
+			final IInvoiceCandDAO.InvoiceableInvoiceCandIdResult invoiceableInvoiceCandId = invoiceCandDAO.getFirstInvoiceableInvoiceCandId(targetOrderId);
+			final InvoiceCandidateId invoiceCandidateId = invoiceableInvoiceCandId.getFirstInvoiceableInvoiceCandId();
+
+			final I_C_Invoice_Candidate invoiceCandidateRecord = invoiceCandDAO.getById(invoiceCandidateId);
+			invoiceCandidateIds.add(invoiceCandidateRecord);
+		}
+
+		//enqueue invoice candidate
+		final List<Integer> t_Selection = invoiceCandidateIds
+				.stream()
+				.map(ic -> ic.getC_Invoice_Candidate_ID())
+				.collect(Collectors.toList());
+
+		final PInstanceId invoiceCandidatesSelectionId = DB.createT_Selection(t_Selection, ITrx.TRXNAME_None);
+
+		final PlainInvoicingParams invoicingParams = new PlainInvoicingParams();
+		invoicingParams.setIgnoreInvoiceSchedule(false);
+		invoicingParams.setSupplementMissingPaymentTermIds(false);
+		invoicingParams.setAssumeOneInvoice(true);
+
+		final IInvoiceCandidateEnqueueResult iInvoiceCandidateEnqueueResult = invoiceCandBL.enqueueForInvoicing()
+																							.setContext(Env.getCtx())
+																							.setFailIfNothingEnqueued(false)
+																							.setInvoicingParams(invoicingParams)
+																							.enqueueSelection(invoiceCandidatesSelectionId);
+
+		final IInvoiceCandBL.IInvoiceGenerateResult result = invoiceCandBL.generateInvoicesFromQueue(Env.getCtx());
+
+		//wait for the invoice to be created
+		for (I_C_Invoice_Candidate candidate : invoiceCandidateIds)
+		{
+			StepDefUtil.tryAndWait(timeoutSec, 500, candidate::isProcessed);
+		}
+
+		final List<OrderLineId> orderLines = new ArrayList<>();
+		for (final I_C_Order order : orders)
+		{
+			final List<I_C_OrderLine> lines = iOrderDAO.retrieveOrderLines(OrderId.ofRepoId(order.getC_Order_ID()));
+			for (I_C_OrderLine ol: lines)
+			{
+				orderLines.add(OrderLineId.ofRepoId(ol.getC_OrderLine_ID()));
+			}
+		}
+
+		final List<InvoiceId> invoices = invoiceDAO.retrieveInvoicesForOrderLineIds(orderLines);
+		assertThat(invoices.size()).isEqualTo(1);
+
+		final I_C_Invoice invoice = InterfaceWrapperHelper.load(invoices.get(0), I_C_Invoice.class);
+
+		final String invoiceIdentifier = DataTableUtil.extractStringForColumnName(table.asMaps().get(0), I_C_Invoice.COLUMNNAME_C_Invoice_ID + "." + StepDefConstants.TABLECOLUMN_IDENTIFIER);
+		invoiceTable.put(invoiceIdentifier,invoice);
+	}
```

Comment:
pls extend & re-use existing `public void generateInvoice0(final int timeoutSec, @NonNull final DataTable table) throws InterruptedException` and drop this one